### DISCUSSION TITLE
Access to Auth State  from the backend

### DISCUSSION THREAD

**Discussion by rmoskal** [pocketbase/pocketbase#5199] [Q&A] [answered]
I've started my pocketbase journey with:

https://github.com/Egor-S/pocketbase-sveltekit-static

When deployed, the sveltekit application is served from the pocketbase static assets directory (ssr is off).  Everything works great, except code in the pb_hooks directory, like route handlers don't have access to the authstate.  "admin" and "authrecord" are always null.

This is how I log into pocketbase from svelte:

```
	try {
		await pb.collection('users').authWithPassword(data.email, data.password);
		await applyAction({ type: 'redirect', status: 303, location: data.next || '/' });
	} catch (e) {
		if (e instanceof ClientResponseError) {
			await applyAction({ type: 'failure', status: 400, data: { message: e.response.message } });
		} else {
			throw e;
		}
	}
```
	
And I watch the authstore for changes like this:

```
pb.authStore.onChange(() => {
	user.set(pb.authStore.model);
	currentUser = pb.authStore.model;
}, true);
```

I must be missing a step.  Any pointers?

**Comment by ganigeorgiev** [comment]
> like route handlers don't have access to the authstate.

Please provide an example how are you sending the client request and how do you check the state in the hooks (what hook?).

> When deployed, the sveltekit application is served from the pocketbase static assets directory 

This means that only the PocketBase binary, hooks, migrations and static assets are deployed on the server, right? If not, are you sure that your code runs only in the browser (ex. where `applyAction` is executed)?

**Reply by ganigeorgiev** [reply]
The easiest way to rule out PocketBase issue is to create a plain HTML file in `pb_public/index.html` and navigate to `http://127.0.0.1:8090`:

```html
<!DOCTYPE html />
<html>
<head>
    <meta charset="utf-8" />
    <title>Test</title>
</head>
<body>
    <script type="module">
        import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@master/dist/pocketbase.es.mjs"

        const pb = new PocketBase("http://127.0.0.1:8090");

        await pb.collection("users").authWithPassword("test@example.com", "1234567890");

        const result = await pb.collection("example").getFullList()
        console.log(result)
    </script>
</body>
</html>
```

**Reply by rmoskal** [reply]
All that works fine but I think I know what's happening. I have a custom endpoint in pb_hooks:

```
routerAdd("GET", "/hello", (c) => {
    let name = c.pathParam("name")
    const admin  = c.get("admin")      // empty if not authenticated as admin
    const record = c.get("authRecord")
    return c.json(200, { "message": {admin, record} })
})
```

A call it from my sveltekit code:

     const foo = await fetch('/hello')
     
 Looking at it today,  I have to imagine that's a naked http request, no headers, tokens etc. Can I pass the credentials along with the fetch request?

**Reply by ganigeorgiev** [reply, marked as answer]
If you want to send the request with raw `fetch` you'll have to manually add the `Authorization` header with the call, ex.:

```js
await fetch('/hello', {
    headers: { "Authorization": pb.authStore.token }
})
```

But a better alternative would be to use the SDK and its generic `send` method:

```js
await pb.send("/hello")
```
(_see [Sending requests to custom routes](https://pocketbase.io/docs/js-routing/#sending-request-to-custom-routes-using-the-sdks)_)

**Reply by rmoskal** [reply]
DOH! RTFM!.

Of course this works great. Sorry for the trouble!
